src/mono/wasm/runtime/library_mono.js

```diff
@@ -338,6 +339,11 @@ var MonoSupportLib = {
 			return exception_obj ;
 		},
 
+		mono_wasm_add_lazy_load_files: function(lazy_loaded_files) {
```

Comment:
Can this get called more than once?

Reply:
Yes, it will be called whenever the user loads a set of assemblies using the `LazyLoaderService` in Blazor. Typically, this  will be called whenever a user navigates to a new page that loads assemblies lazily.

Any reservations about this?

Reply:
Then should we clear `MONO.lazy_loaded_files` in `mono_wasm_get_lazy_loaded_files`, or when the existing list has been loaded?

Reply:
We can, although I'm not sure it is totally necessary. The flow would look something like this.

1. User navigates to page A.
2. Blazor app loads assemblies lazily.
3. Blazor app calls `mono_wasm_add_lazy_load_files`.
4. MONO.lazy_load_files is set to `lazy_loaded_files` parameter.
5. `mono_wasm_add_lazy_load_files` invokes `debugger;` to emit Debugger.paused event.
6. Proxy issues `mono_wasm_get_lazy_loaded_files` to get assemblies and loads them into the DebugStore.
7. User navigates to page B.
8. Blazor app loads another set of lazily-loaded assemblies.
7. Blazor app calls `mono_wasm_add_lazy_load_files` and the MONO.lazy_loaded_files list is overriden again.

So, I think it should be fine, but let me know if I am missing a subtlety here.

Reply:
Silly question: When a user navigates to page B, it's like a new page was loaded, right? So, the runtime will be re-initialized, assemblies will be loaded etc? If so, then `mono_wasm_add_lazy_load_files` won't get called twice while in the same .. "session"?

Reply:
> When a user navigates to page B, it's like a new page was loaded, right? So, the runtime will be re-initialized, assemblies will be loaded etc? 

The navigation event occurs via a `history.pushState` event in the browser. The URL changes and Blazor reacts to that change and invokes the handlers that trigger the loading of assemblies. In this case, the runtime won't be re-initialized.

Reply:
ok, so the state in `MONO`, would remain the same. Ok, so it might have stale list of files, but if we are only ever looking at it when paused due to this function being invoked, then we should be fine. Ok, understood.